### PR TITLE
Update Release CRD schema

### PR DESCRIPTION
**Pull request by mcharriere**
Signed-off-by: Matias Charriere <matias@giantswarm.io>

### CODE REVIEW

File: master/release-release-v1alpha1.json [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
     "kind": {
-      "description": "Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds",
+      "description": "Kind is a string value representing the REST resource this object represents.\nServers may infer this from the endpoint the client submits requests to.\nCannot be updated.\nIn CamelCase.\nMore info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds",
       "type": "string"
     },
     "metadata": {
       "properties": {
         "name": {
-          "pattern": "^v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(-[\\.0-9a-zA-Z]*)?$",
+          "pattern": "^(((aws|azure|vsphere|cloud-director|eks)-)|v)(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(-[\\.0-9a-zA-Z]*)?$",
```
Comment (AverageMarcus) [inline master/release-release-v1alpha1.json:15]: What is this exactly? I'm surprised not to see `capa` here.

Comment (AverageMarcus) [reply master/release-release-v1alpha1.json:15]: Oh, ignore me, `capa` is only used in the directory structure, not the release.yaml name 🤦  Not at all confusing.

**Review by AverageMarcus — DISMISSED** [review]

File: master/release-release-v1alpha1.json (hunk 1 above)
Comment (AverageMarcus) [reply master/release-release-v1alpha1.json:15]: Oh actually, looks like the `v` is meant to be optional though? e.g. https://github.com/giantswarm/releases/blob/3bc4ab8a804fffff626d121d17bae62dee5ee23e/capa/v29.1.0/release.yaml#L4

Comment (mcharriere) [reply master/release-release-v1alpha1.json:15]: https://github.com/giantswarm/releases/blob/master/capa/v29.1.0/release.yaml#L4

actually, for `capa` the prefix is `aws`. This update has been taken from the release CRD in `golem` (using the sync script in this repo).

Comment (AverageMarcus) [reply master/release-release-v1alpha1.json:15]: Yeah, I think my last comment still stands though right?

Comment (mcharriere) [reply master/release-release-v1alpha1.json:15]: if I read that pattern right, it's either the `provider` prefix or the `v`. So it's not optional: it's mandatory on Vintage and not expected on CAPI.

Comment (AverageMarcus) [reply master/release-release-v1alpha1.json:15]: Oh gotcha! Regex fun! :p

**Review by AverageMarcus — APPROVED** [review]

**Pull request merged**